content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,608 @@
+            embeddings.append(self.embedding_model[word])
+        return embeddings
+     
+embedding_vectorizer = EmbeddingVectorizer()
+```
+
+We also need to apply a similar operation to the address tags (e.g. StreeNumber, StreetName). 
+This time, however, the `vectorizer` needs to convert the tags into categorical values (e.g. StreetNumber -> 0). 
+
+For simplicity, we will use a `DatasetVectorizer` class that will apply the vectorizing process using both 
+the embedding and the address vectorize process that we just describe.
+
+```python
+class DatasetVectorizer:
+    def __init__(self, embedding_vectorizer):
+        self.embedding_vectorizer = embedding_vectorizer
+        self.tags_set = {
+            "StreetNumber": 0,
+            "StreetName": 1,
+            "Unit": 2,
+            "Municipality": 3,
+            "Province": 4,
+            "PostalCode": 5,
+            "Orientation": 6,
+            "GeneralDelivery": 7
+        }
+
+    def vectorize(self, data):  # We vectorize inplace
+        for idx, item in enumerate(data):
+            data[idx] = self._item_vectorizing(item)
+
+    def _item_vectorizing(self, item):
+        address = item[0]
+        address_vector = self.embedding_vectorizer(address)
+
+        tags = item[1]
+        idx_tags = self._convert_tags_to_idx(tags)
+
+        return address_vector, idx_tags
+
+    def _convert_tags_to_idx(self, tags):
+        idx_tags = []
+        for tag in tags:
+            idx_tags.append(self.tags_set[tag])
+        return idx_tags
+
+dataset_vectorizer = DatasetVectorizer(embedding_vectorizer)
+```  
+
+Now, let's vectorize our dataset.
+
+```python
+dataset_vectorizer.vectorize(train_data)
+dataset_vectorizer.vectorize(valid_data)
+dataset_vectorizer.vectorize(test_data)
+```
+
+```python
+train_data[0]
+```
+
+> Here a example after the vectorizing process
+
+```
+([array([ 5.30934185e-02, -6.72420338e-02,  8.96735638e-02, -3.26771051e-01,
+         -4.42410737e-01, -1.32014668e-02,  1.50324404e-01, -2.50251926e-02,
+          1.24011010e-01, -1.68488681e-01,  4.80616689e-02, -5.40233105e-02,
+          1.21191796e-02,  3.89859192e-02,  1.25505164e-01, -1.40419468e-01,
+         -2.62053646e-02,  1.01731330e-01, 
+         ...
+         -5.71550950e-02, -3.92134525e-02,  4.85491045e-02,  4.82993454e-01,
+          3.35614271e-02, -5.97143888e-01, -9.82549414e-02,  8.23293403e-02],
+        dtype=float32), #dimension of 300 # first word
+        ... 
+        # word N of the sequence
+        array([-6.28125593e-02, -1.72182580e-03,  1.27990674e-02,  7.47001171e-02,
+        ...
+        3.74843590e-02],
+        dtype=float32)],
+    
+ #second element of the tuple
+ [0, 1, 1, 1, 3, 4, 5, 5] #the ground truth tag)
+```
+
+### DataLoader
+> We use a first trick, ``padding``.
+
+Now, since the addresses are not all of the same size, it is impossible to batch them together; recall that all tensor elements must have the same lengths. But there is a trick, padding!
```

Comment:
```suggestion
Now, because the addresses are not all of the same size, it is impossible to batch them together; recall that all tensor elements must have the same lengths. But there is a trick, padding!
```